### PR TITLE
FD-38-B replaced FD-38: add post badges endpoint, tests are passing

### PR DESCRIPTION
**Pull request by iaroslavm**

### CODE REVIEW

File: ferrilata-devilline/Controllers/ApiControllers/BadgesController.cs [hunk 1]
```diff
@@ -22,5 +23,19 @@ public IActionResult GetBadges()
         {
             return Ok(_badgeService.GetAllDTO());
         }
+
+        [HttpPost]
+        [Route("/api/post/badges")]
+        public IActionResult PostBadge([FromBody] BadgeInDTO IncomingBadge)
+        {
+
```
Comment (AlexTeper) [inline ferrilata-devilline/Controllers/ApiControllers/BadgesController.cs]: Empty line

File: ferrilata-devilline/Repositories/BadgeRepository.cs [hunk 2]
```diff
@@ -40,5 +40,46 @@ public void DeleteBadgeById(long id)
...
             _applicationContext.SaveChanges();
         }
+
+        public void SaveOrUpdateBadge(Badge badge)
+        {
+            if (_applicationContext.Badges.Find(badge.BadgeId) == null)
+            {
+                _applicationContext.Badges.Add(badge);
+            }
+            else
```
Comment (AlexTeper) [inline ferrilata-devilline/Repositories/BadgeRepository.cs]: Just remove the else as we discussed

File: ferrilata-devilline/Repositories/BadgeRepository.cs [hunk 3]
```diff
@@ -40,5 +40,46 @@ public void DeleteBadgeById(long id)
...
+
+        public void SaveOrUpdateLevel(Level level)
+        {
+            if (_applicationContext.Levels.Find(level.LevelId) == null)
+            {
+                _applicationContext.Levels.Add(level);
+            }
+            else
+            {
+                var oldLevel = FindLevelById(level.LevelId);
```
Comment (AlexTeper) [inline ferrilata-devilline/Repositories/BadgeRepository.cs]: Same here

**Review by AlexTeper — CHANGES_REQUESTED** [review]

File: ferrilata-devilline/Controllers/ApiControllers/BadgesController.cs (hunk 1 above)
Comment (iaroslavm) [reply ferrilata-devilline/Controllers/ApiControllers/BadgesController.cs]: done

File: ferrilata-devilline/Repositories/BadgeRepository.cs (hunk 2 above)
Comment (iaroslavm) [reply ferrilata-devilline/Repositories/BadgeRepository.cs]: done

File: ferrilata-devilline/Repositories/BadgeRepository.cs (hunk 3 above)
Comment (iaroslavm) [reply ferrilata-devilline/Repositories/BadgeRepository.cs]: done

**Review by sabyanna — APPROVED** [review]

**Pull request merged**